### PR TITLE
Add metadata to some filters

### PR DESCRIPTION
**Pull request by MarikaChlebowska**
# Description
This PR preserves metadata when running some filters. As discussed on discord that helps when running for example `ls | reject modified` because it keeps colouring and links.

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-command/src/filters/drop/column.rs [hunk 1]
```diff
@@ -121,14 +123,18 @@ fn drop_cols(
                             drop_record_cols(val, head, &drop_cols)?
                         }
                     }
-                    Ok(Value::list(vals, span).into_pipeline_data())
+                    Ok(Value::list(vals, span)
+                        .into_pipeline_data()
```
Comment (fdncred) [inline crates/nu-command/src/filters/drop/column.rs:127]: I think ones like this should probably be `.into_pipeline_data_with_metadata(metadata)`

**Review by fdncred — CHANGES_REQUESTED** [review]
Generally, this looks good. There's just a place or two that it needs to be tweaked. Thanks!

File: crates/nu-command/src/filters/drop/column.rs (hunk 1 above)
Comment (MarikaChlebowska) [reply crates/nu-command/src/filters/drop/column.rs:127]: I did it like that at first but then I saw it being done the other way in few places so I changed it for consistency. I can change it back

Comment (fdncred) [reply crates/nu-command/src/filters/drop/column.rs:127]: ok. Where possible, it should be `.into_pipeline_data_with_metadata(metadata)` but in some places it isn't possible.

**Review by fdncred — APPROVED** [review]

**Comment by fdncred** [comment]
nice work!

**Comment by fdncred** [comment]
Let's go! I've been waiting for someone to pick up this task for so long. Really appreciate you and this PR!

**Pull request merged**